### PR TITLE
Log retention of curve cluster service

### LINKED ISSUE #197: [request help] please rotate service log

### PR DESCRIPTION
**Pull request by hdbdn77**
fix #197
Must be executed on the working node

### CODE REVIEW

File: internal/tools/log_rotate.sh [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+}
+
+############################ FUNCTIONS
+load_config() {
+    # Specify JSON file path
+    json_file="config.json"
+
+    # Check if the file exists
+    if [ ! -f "$json_file" ]; then
+        die "json file ${json_file} does not exist.\n"
```
Comment (caoxianfei1) [inline internal/tools/log_rotate.sh:29]: is ok？ die and success function seems to require two parameter.

File: internal/tools/log_rotate.sh [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    compress_or_clean=$(jq -r '.compressOrClean' "$json_file")
+    max_tar_file=$(jq -r '.maxTarFile' "$json_file")
+
+}
+
+precheck() {
+    # Check whether interval is greater than or equal to 0
+    if [[ ! ($interval =~ ^[0-9]+$) ]]; then
+        die "The interval parameter is invalid.\n"
+    elif [[ ! ($interval -ge 0) ]]; then
```
Comment (caoxianfei1) [inline internal/tools/log_rotate.sh:44]: need to convert into integer and then determine?

File: internal/tools/log_rotate.sh [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    absolute_path_regex="^\/([a-zA-Z0-9-]+\/)*[a-zA-Z0-9-]+\/?$"
+    if ! [[ $log_directory =~ $absolute_path_regex ]]; then
+        die "The log_directory parameter is invalid.\n"
+    fi
+
+
+     # Check if compress_or_clean is equal to 0 or equal to 1
+    if [[ ! ($compress_or_clean =~ ^[0-9]+$) ]]; then
+        die "The compress_or_clean parameter is invalid.\n"
+    elif [[ ! ($compress_or_clean -eq 0 || $compress_or_clean -eq 1) ]]; then
```
Comment (caoxianfei1) [inline internal/tools/log_rotate.sh:58]: Ditto.

File: internal/tools/log_rotate.sh [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        die "The compress_or_clean parameter is invalid.\n"
+    fi
+
+    # Check if max_tar_file is greater than or equal to 1
+    if [[ ! ($max_tar_file =~ ^[0-9]+$) ]]; then
+        die "The max_tar_file parameter is invalid.\n"
+    elif [[ ! ($max_tar_file -ge 1) ]]; then
```
Comment (caoxianfei1) [inline internal/tools/log_rotate.sh:65]: Ditto.

File: internal/tools/log_rotate.sh (hunk 1 above)
Comment (hdbdn77) [reply internal/tools/log_rotate.sh:29]: 如果你调用 die 函数时只提供了一个参数，那么 ${2} 就是空的。

**Pull request closed without merging**